### PR TITLE
proxy: Generate a label enum (with nice formatting) for Unix error codes

### PR DESCRIPTION
**Pull request by hawkw**
This branch adds an enum of human-readable metric labels for 
Unix error codes that displays them using the names given in
`errno.h`. This isn't currently used, but the enum (and the
macro that produces it) will be used in subsequent branches.

### CODE REVIEW

**Comment by hawkw** [comment]
@seanmonstar `io::Error` isn't `Clone`, so we can't currently put it in events --- we'd have to just send the `ErrorKind`, which lacks a `Display` implementation. We could use the `Debug` formatter on `ErrorKind`, which I had considered previously, but I thought it was better to produce a less Rust-specific representation.

Also, I think we generally prefer the metric labels to be relatively short, and we want to imply that there's a fixed set of possible values, which I think using Unix error names does better than using the short sentences produced by `io::Error`s `Display` implementation. I think it's correct to use that formatter in log messagews, but it's not quite what we'd want for labels.

Just my 2¢!

**Comment by seanmonstar** [comment]
@hawkw that's fair! I think it'd be wrong in Windows, right? (Not that we're currently targeting Windows, but it's also not off the table.)

**Comment by hawkw** [comment]
@seanmonstar That's a good point. Perhaps on Windows we should just skip the enum and report the OS error codes --- it would be nice to not report something completely incorrect, but I also don't think supporting Windows is a high enough priority for it to be worth adding a separate mapping of error codes to names for Windows (esp. since I suspect most Windows-based developers would be running Conduit in WSL anyway?).

File: proxy/src/telemetry/metrics/labels.rs [hunk 1]
```diff
@@ -87,6 +120,146 @@ pub struct DstLabels {
...
+        26 => ETXTBSY,   // Text file busy
+        27 => EFBIG,     // File too large
+        28 => ENOSPC,    // No space left on device
+        29 => ESPIPE,    // Illegal seek
+        30 => EROFS,     // Read-only file system
+        31 => EMLINK,    // Too many links
+        32 => EPIPE,     // Broken pipe
+        33 => EDOM,      // Math argument out of domain of func
+        34 => ERANGE,    // Math result not representable
+        35  => EDEADLK        ,  // Resource deadlock would occur
```
Comment (olix0r) [inline proxy/src/telemetry/metrics/labels.rs]: what's going on with this formatting

Comment (hawkw) [reply proxy/src/telemetry/metrics/labels.rs]: sorry about that --- fixed in 850d773b83b8547aa7314944721c7e656789baf8

**Pull request closed without merging**